mne/io/curry/curry.py

```diff
@@ -0,0 +1,237 @@
+
+            if re.match("LABELS.*? START_LIST", line):
+                save_labels = True
+
+            # if "SENSORS START_LIST" in line:
+            if re.match("SENSORS.*? START_LIST", line):
+                save_ch_pos = True
+
+    ch_pos = np.array(ch_pos, dtype=float)
+    # TODO find a good method to set montage (do it in read_montage instead?)
+
+    info = create_info(ch_names, sfreq)
+
+    # TODO; There's still a lot more information that can be brought into info["chs"]. However i'm not sure what to do with MEG chans here
+    for ch_dict in info["chs"]:
+        ch_dict["cal"] = CAL
+
+    return info, n_trials, n_samples, curry_vers, data_format
+
+
+def read_raw_curry(input_fname, preload=False):
+    """
+    Read raw data from Curry files.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to a curry file with extensions .dat, .dap, .rs3, .cdt, cdt.dpa,
+        .cdt.cef or .cef.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory). If the curry file
+        is stored in ASCII data format, then preload must be `True`.
+
+    Returns
+    -------
+    raw : instance of RawCurry
+        A Raw object containing CURRY data.
+
+    """
+
+    DATA_FILE_EXTENSION = {7: '.dat', 8: '.cdt'}
+
+    # we don't use os.path.splitext to also handle extensions like .cdt.dpa
+    fname_base, ext = input_fname.split(".", maxsplit=1)
+
+    curry_vers = _get_curry_version(ext)
+    _check_missing_files(input_fname, fname_base, curry_vers)
+
+    info, n_trials, n_samples, curry_vers, data_format = _read_curry_info(fname_base, curry_vers)
+    events = _read_curry_events(fname_base, curry_vers)
+    info["events"] = events
+
+    raw = RawCurry(fname_base + DATA_FILE_EXTENSION[curry_vers], info, n_samples, data_format)
+
+    return raw
+
+
+class RawCurry(BaseRaw):
+    """"""
+
+    def __init__(self, data_fname, info, n_samples, data_format, montage=None, eog=(), ecg=(),
+                 emg=(), misc=(), preload=False, verbose=None):  # noqa: D102
+
+        data_fname = os.path.abspath(data_fname)
+
+        last_samps = [n_samples - 1]
+
+        if preload == False and data_format == "ASCII":
+            warn('Got ASCII format data as input. Data will be preloaded.')
+
+
+        super(RawCurry, self).__init__(
+            info, preload, filenames=[data_fname], last_samps=last_samps, orig_format='int',
+            verbose=verbose)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a chunk of raw data."""
+
+        _read_segments_file(self, data, idx, fi, start, stop, cals, mult, dtype="float32")
```

Comment:
this is never called

Reply:
the `_read_segments_file` part is not called? it surely has to be, since fails the tests if you e.g. do
```python
_read_segments_file(self, data, idx, fi, start, stop, cals, mult, dtype="int32")
```
but not for
```python
_read_segments_file(self, data, idx, fi, start, stop, cals, mult, dtype="float32")
```
i adapted this from the RawNicolet BTW, it's working there too

Reply:
It appeared as not cover. Don't bother for the moment (but do not resolve the conversation). keep it as a reminder to make sure that is properly cover etc.